internal/bufferedwrites/upload_handler_test.go

```diff
@@ -111,46 +115,56 @@ func (t *UploadHandlerTest) TestUpload_CreateObjectWriterFails() {
 
 func (t *UploadHandlerTest) TestFinalizeWithWriterAlreadyPresent() {
 	writer := &storagemock.Writer{}
-	writer.On("Close").Return(nil)
+	mockObj := &gcs.Object{}
+	t.mockBucket.On("FinalizeUpload", mock.Anything, writer).Return(mockObj, nil)
 	t.uh.writer = writer
 
-	err := t.uh.Finalize()
+	obj, err := t.uh.Finalize()
 
-	assert.NoError(t.T(), err)
+	require.NoError(t.T(), err)
+	require.NotNil(t.T(), obj)
+	assert.Equal(t.T(), mockObj, obj)
 }
 
 func (t *UploadHandlerTest) TestFinalizeWithNoWriter() {
 	writer := &storagemock.Writer{}
 	t.mockBucket.On("CreateObjectChunkWriter", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(writer, nil)
 	assert.Nil(t.T(), t.uh.writer)
-	writer.On("Close").Return(nil)
+	mockObj := &gcs.Object{}
+	t.mockBucket.On("FinalizeUpload", mock.Anything, writer).Return(mockObj, nil)
 
-	err := t.uh.Finalize()
+	obj, err := t.uh.Finalize()
 
-	assert.NoError(t.T(), err)
+	require.NoError(t.T(), err)
+	require.NotNil(t.T(), obj)
+	assert.Equal(t.T(), mockObj, obj)
 }
 
 func (t *UploadHandlerTest) TestFinalizeWithNoWriter_CreateObjectWriterFails() {
 	t.mockBucket.On("CreateObjectChunkWriter", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil, fmt.Errorf("taco"))
 	assert.Nil(t.T(), t.uh.writer)
 
-	err := t.uh.Finalize()
+	obj, err := t.uh.Finalize()
 
 	require.Error(t.T(), err)
 	assert.ErrorContains(t.T(), err, "taco")
 	assert.ErrorContains(t.T(), err, "createObjectWriter")
+	assert.Nil(t.T(), obj)
 }
 
-func (t *UploadHandlerTest) TestFinalize_WriterCloseFails() {
+func (t *UploadHandlerTest) TestFinalize_FinalizeUploadFails() {
```

Comment:
some tests we are having _ and others not. Lets be consistent and not use _ anywhere.